### DISCUSSION TITLE
Changing Expiry Times

### DISCUSSION THREAD

**Discussion by goncalopolido** [PrivateBin/PrivateBin#1188] [Q&A] [answered]
I am currently running a PrivateBin instance using Docker, specifically the privatebin/nginx-fpm-alpine image from Docker Hub ([privatebin/nginx-fpm-alpine](https://hub.docker.com/r/privatebin/nginx-fpm-alpine/)). I launched the instance with the following command:

`docker run -d --restart="always" --read-only -p 8080:8080 -v $PWD/privatebin-data:/srv/data privatebin/nginx-fpm-alpine`

I want to customize the expiry times for self-destructing data in my PrivateBin instance. By default, PrivateBin has a predefined expiry time for pastes, but I'd like to set a specific time that suits my needs. In particular, I'd like to set a shorter expiry time or simply remove the 1 week 1 month 1 year and never expiry times, those are basically the ones that i don't want to have enabled.
 
Could someone please guide me through the process of changing the expiry times in a PrivateBin Docker instance? Are there specific configurations or settings I should modify in the Docker container, and if so, what steps should I follow to make this adjustment?

Any assistance or insights on this matter would be greatly appreciated. Thank you in advance!

![image](https://github.com/PrivateBin/PrivateBin/assets/83615960/cb45b30f-ccc4-4fde-b3ec-fb23d4d8d4d3)

**Comment by rugk** [comment, marked as answer]
You need to mount a custom options file. See https://github.com/PrivateBin/docker-nginx-fpm-alpine?tab=readme-ov-file#custom-configuration

There, you need to adjust the following options:
https://github.com/PrivateBin/PrivateBin/wiki/Configuration#expiration

**Comment by goncalopolido** [comment]
so i have to stop and delete the current container right?

**Reply by rugk** [reply]
Stop, adjust and restart with the new options/parameters, at least. That's always the recommend practice.

Generally https://docker-curriculum.com/ may be helpful.

**Reply by goncalopolido** [reply]
I have a question regarding the conf.php file. As per my understanding, conf.php should be a file, but it seems to be recognized as a directory within the container. Am I overlooking something, or is there an issue with the configuration?

```
root@privatebin:/home# ls
root@privatebin:/home#
root@privatebin:/home# docker run -d --restart="always" --read-only -p 8080:8080 -v $PWD/conf.php:/srv/cfg/conf.php:ro -v $PWD/privatebin-data:/srv/data privatebin/nginx-fpm-alpine
7e773f957ab9c9b6dc83a7860fbc51c53ee79041960e00c37973c80e4ad27d36
root@privatebin:/home# docker ps
CONTAINER ID   IMAGE                         COMMAND                  CREATED              STATUS              PORTS                                       NAMES
7e773f957ab9   privatebin/nginx-fpm-alpine   "/etc/init.d/rc.local"   About a minute ago   Up About a minute   0.0.0.0:8080->8080/tcp, :::8080->8080/tcp   mystifying_driscoll
root@privatebin:/home# ls
conf.php  privatebin-data
root@privatebin:/home# cd conf.php
root@privatebin:/home/conf.php#
root@privatebin:/home/conf.php# ls
root@privatebin:/home/conf.php#
root@privatebin:/home/conf.php# cd ..
root@privatebin:/home#
root@privatebin:/home# docker exec -it 7e773f957ab9 /bin/sh
/var/www $
/var/www $ cd /srv/cfg
/srv/cfg $ 
/srv/cfg $ ls
conf.php
/srv/cfg $ cd conf.php
/srv/cfg/conf.php $ 
/srv/cfg/conf.php $ ls
/srv/cfg/conf.php $ 
```

**Reply by elrido** [reply]
If you ask docker to attach a volume and pass it a non-existing location (your ls shows you have no conf.php outside the container), docker will automatically create that location as a directory for you and attach that.

You'd need to stop the container, delete the directory docker created, create the file in its place, ensure it will be readable for the services in the container, then re-create the container. You should now find a read-only file mapped into the container instead of a directory.
